implementation/e2e_protection/include/e2e/profile/profile05/profile_05.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2020-2021 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef VSOMEIP_V3_E2E_PROFILE05_PROFILE05_HPP
+#define VSOMEIP_V3_E2E_PROFILE05_PROFILE05_HPP
+
+#include <cstdint>
+
+#include <vsomeip/defines.hpp>
+
+#include "../../../buffer/buffer.hpp"
+
+// The MSB of the dataID is the instance identifier.
+// Therefore, the instance identifier must fit into a single byte.
+#define VSOMEIP_E2E_PROFILE05_MAX_INSTANCE 0x00ff
+
+namespace vsomeip_v3 {
+namespace e2e {
+namespace profile05 {
+
+struct profile_config;
+
+class profile_05 {
+public:
+    static uint16_t compute_crc(const profile_config &_config, const e2e_buffer &_buffer);
+};
+
+struct profile_config {
+    profile_config() = delete;
+
+    profile_config(uint32_t _data_id, size_t _offset,
```

Comment:
Profile 05 does not require _min_data_length nor _max_data_length. Instead, it uses a fixed data length. The necessary parameters can be found [here](https://www.autosar.org/fileadmin/standards/R20-11/FO/AUTOSAR_PRS_E2EProtocol.pdf#chapter.8)